### PR TITLE
[flink] refactor update data fields process function

### PR DESCRIPTION
**Pull request by zhangjun0x01**
### Purpose

Extract some common code

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/MultiTableUpdatedDataFieldsProcessFunction.java [hunk 1]
```diff
@@ -53,16 +47,17 @@
...
             LoggerFactory.getLogger(MultiTableUpdatedDataFieldsProcessFunction.class);
 
     private final Map<Identifier, SchemaManager> schemaManagers = new HashMap<>();
-    private Catalog catalog;
+
     private final Catalog.Loader catalogLoader;
 
     public MultiTableUpdatedDataFieldsProcessFunction(Catalog.Loader catalogLoader) {
+        super(catalogLoader);
         this.catalogLoader = catalogLoader;
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/MultiTableUpdatedDataFieldsProcessFunction.java:61]: delete `this.catalogLoader = catalogLoader`

**Review by yuzelin — COMMENTED** [review]
I left some comments, please take a look.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/MultiTableUpdatedDataFieldsProcessFunction.java (hunk 1 above)
Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/MultiTableUpdatedDataFieldsProcessFunction.java]: thanks for your suggestion, addressed it

**Review by yuzelin — COMMENTED** [review]
+1

**Pull request merged**